app/controllers/file_controller.rb

```diff
@@ -50,7 +50,9 @@ def show
     disposition = params[:disposition] || 'inline'
     filename = params["filename"] || "#{objid}#{fileid}"
     headers['Content-Disposition'] = "#{disposition}; filename=#{filename}"
-    if ds.mimeType
+    if filename.include?('.jpg')
```

Comment:
I think "application/octet-stream" is coming from the jpeg file itself unless the metadata record is corrupted., which was extracted by Jhove. If the format of a file can't be identified by Jhove, it will give it a mime type like "application/octet-stream" and a format like "bytestream".